### ISSUE TITLE
Support for composite partition keys in Kundera-Cassandra

### ISSUE THREAD

**Issue by capalbc** [Impetus/kundera#550]
CQL3 allows for composite partition keys, for example:

```
cqlsh:test> CREATE TABLE example ( 
        ... partitionKey1 text,
        ... partitionKey2 text,
        ... clusterKey1 text,
        ... clusterKey2 text,
        ... normalField1 text,
        ... normalField2 text,
        ... PRIMARY KEY (
                (partitionKey1, partitionKey2), 
                clusterKey1, clusterKey2
              )
        ... );
```

insert data:

```
cqlsh:test> INSERT INTO example (
        ... partitionKey1, 
        ... partitionKey2, 
        ... clusterKey1, 
        ... clusterKey2, 
        ... normalField1, 
        ... normalField2
        ... ) VALUES ( 
        ... 'partitionVal1', 
        ... 'partitionVal2', 
        ... 'clusterVal1', 
        ... 'clusterVal2', 
        ... 'normalVal1', 
        ... 'normalVal2'); 
```

query:

```
cqlsh:test> SELECT * FROM example;    
 partitionkey1 | partitionkey2 | clusterkey1 | clusterkey2 | normalfield1 | normalfield2
---------------+---------------+-------------+-------------+--------------+--------------
 partitionVal1 | partitionVal2 | clusterVal1 | clusterVal2 |   normalVal1 |   normalVal2
```

cassandra-cli:

```
[default@test] list example;
-------------------
RowKey: partitionVal1:partitionVal2
=> (column=clusterVal1:clusterVal2:, value=, timestamp=1374630892473000)
=> (column=clusterVal1:clusterVal2:normalfield1, value=6e6f726d616c56616c31, timestamp=1374630892473000)
=> (column=clusterVal1:clusterVal2:normalfield2, value=6e6f726d616c56616c32, timestamp=1374630892473000)
```

It would be nice if this was supported for in Kundera-Cassandra.

Perhaps the @Embedded annotation can be used to denote the composite partition key, for example:

```
@Entity
public class Example
{
    @EmbeddedId
    private CompoundPrimaryKey primaryKey;

    @Column private String normalField1; 
    @Column private String normalField2;
    ...
```

```
@Embeddable
public class CompoundPrimaryKey
{
    @Embedded
    private CompositePartitionKey partitionKey;

    @Column private String clusterKey1; 
    @Column private String clusterKey2;
    ...
```

```
@Embeddable
public class CompositePartitionKey
{
    @Column private String partitionKey1; 
    @Column private String partitionKey2;
    ...
```

**Comment by mevivs** [comment]
Hi,
Yes. It's very much on plan. Will post an update for this.

-Vivek

**Comment by capalbc** [comment]
Wonderful thanks!  Yes, please do.

Reading about the inclusion of support for the Datastax java-driver in the upcoming 2.11 release (#385), I'm very interested in the workings of Kundera and using it in my projects.  So in the meantime I'll pull the Kundera code and try to familiarize myself with it.  If you can point me to the main classes at play with this enhancement, I may even be able to help contribute with development.

**Comment by chhavigangwal** [comment]
Fixed in Kundera-2.11.1
